File: src/tests/Mappings/First.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace Extender
{

      [Serializable()]
		public class First : TableMetadata
		{

                   public enum FirstFields
                   {
                      Id,
                      Name
                  }


			    private DatabaseField[] _fields;

		    	public First()
			    {
					    _fields = new DatabaseField[2];
                    _fields[0] = new DatabaseField(DbType.Int32,"Id",true,true,null);
                    _fields[1] = new DatabaseField(DbType.String,"Name",false,false,null);
 
                        this.currentTableName = "First";

this.listRelations.Add(new ParentTableRelation("Second", "FirstId",TableRelationCardinality.OneToMany,true));

                  }


			public override DatabaseField[] TableFields 
			{
				get{ return _fields;}
				set{_fields = value;}
			}
          public First Clone()
          {
                 return this.Clone<First>();
          }

public System.Int32? Id
{
    get
    {
          return (System.Int32? ) (this.GetField("Id")).fieldValue;
    }

    set
    {
          this.SetFieldValue("Id", value);
    }
}


public System.String Name
{
    get
    {
          return (System.String) (this.GetField("Name")).fieldValue;
    }

    set
    {
          this.SetFieldValue("Name", value);
    }
}

public Array GetSecond()
{
    Type relatedType = Type.GetType("Extender.Second");
    return this.GetRelatedTableData(relatedType);
}
}
}

File: src/tests/MappingsVB/First.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock

Namespace Extender
 


		Public Class First
		             Inherits TableMetadata
			Private _fields() as DatabaseField

			Sub New() 
			
					_fields = new DatabaseField(2) {}
 _fields(0) = new DatabaseField(DbType.Int32,"Id",True,True,Nothing)
 _fields(1) = new DatabaseField(DbType.String,"Name",False,False,Nothing)
 
me.currentTableName = "First"

Me.listRelations.Add(new ParentTableRelation("Second", "FirstId",TableRelationCardinality.OneToMany,True))

End Sub



			Public Overrides Property TableFields() as DatabaseField() 
		   	   Get
				  Return _fields
			   End Get
			   Set (ByVal Value as DatabaseField())
			      _fields = value
			   End Set
			End Property

     Public Property Id As System.Int32
        Get
           Return  CType((Me.GetField("Id")).fieldValue,System.Int32)
        End Get 
        Set(ByVal Value As System.Int32)
          Me.SetFieldValue(Id, Value)
        End Set
      End Property

     Public Property Name As System.String
        Get
           Return  CType((Me.GetField("Name")).fieldValue,System.String)
        End Get 
        Set(ByVal Value As System.String)
          Me.SetFieldValue(Name, Value)
        End Set
      End Property
Public Function GetSecond() As Array
    Dim relatedType As Type = Type.GetType("Extender.Second")
    Return Me.GetRelatedTableData(relatedType)
End Function
End Class
End Namespace
